scripts/build_llvm.sh

```diff
@@ -74,16 +74,13 @@ fi
 if [ "$llvm_source" = "" ]; then
   echo "Cloning LLVM submodule..."
   cd "$this_file_dir" && cd $(git rev-parse --show-toplevel)
-  llvm_source=~/.llvm-project
-  llvm_repo="$(git config --file=.gitmodules submodule.tpls/llvm.url)"
-  llvm_commit="$(git submodule | grep tpls/llvm | cut -c2- | cut -d ' ' -f1)"
-  git clone --filter=tree:0 "$llvm_repo" "$llvm_source"
-  cd "$llvm_source" && git checkout $llvm_commit
+  llvm_source=$(pwd)/tpls/llvm
+  git submodule update --init --recommend-shallow --single-branch tpls/llvm
```

Comment:
shallow doesn't work well to cherry-pick commits. `git clone --filter=tree:0` avoids the limitations of a shallow repo clone and is equally as fast.

Reply:
In my opinion, this is not the best way to do this. The llvm-project repo is massive and constantly cloning it live in a number of different CI tasks is not a good idea.

First, everything we do should use the llvm-project release/16.x branch (which is the SHA on tpls/llvm). This avoids any issues from using arbitrary llvm versions for disparate parts of our project. This never changes as we're stuck with the limitations of QIR on 16.x.

Second, the subprojects and other prerequisites should be downloaded as part of a standard "development" image build. This image would only be rebuilt when the prerequisites change. It should be a known, published image and not dependent on caches which may be cleared out from underneath the CI.

Finally, instead of cherry-picking SHAs from future branches like this, we should just have a diff file with the patches and apply those changes to the top of the branch. We already do this for pybind11. Applying the patch from a diff file avoids downloading an entire copy of llvm every time which saves bandwidth and time. It's also trivially reversible.